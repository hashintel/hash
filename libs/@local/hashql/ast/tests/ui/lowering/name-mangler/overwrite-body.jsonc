//@ run: pass
//@ description: The name should only be mangled in the body of the expression, not the value assignment
["let", "a", { "#literal": 2 }, ["let", "a", "a", "a"]]
