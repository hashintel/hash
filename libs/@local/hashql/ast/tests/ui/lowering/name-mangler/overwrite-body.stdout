Expr#4294967040@27
  ExprKind (Let)
    LetExpr#4294967040@27 (name: a:0)
      Expr#4294967040@9
        ExprKind (Literal)
          LiteralExpr#4294967040@8
            LiteralKind (Integer)
              IntegerLiteral (2)
      Expr#4294967040@26
        ExprKind (Let)
          LetExpr#4294967040@26 (name: a:1)
            Expr#4294967040@21
              ExprKind (Path)
                Path#4294967040@21 (rooted: false)
                  PathSegment#4294967040@20 (name: a:0)
            Expr#4294967040@25
              ExprKind (Path)
                Path#4294967040@25 (rooted: false)
                  PathSegment#4294967040@24 (name: a:1)
